pkg/api/resource_helpers.go

```diff
@@ -74,6 +76,24 @@ func GetExistingContainerStatus(statuses []ContainerStatus, name string) Contain
 	return ContainerStatus{}
 }
 
+// IsPodAvailable returns true if a pod is available; false otherwise.
+// Precondition for an available pod is that it must be ready. On top
+// of that, there are two cases when a pod can be considered available:
+// 1. minReadySeconds == 0, or
+// 2. LastTransitionTime (is set) + minReadySeconds < current time
+func IsPodAvailable(pod *Pod, minReadySeconds int32, now unversioned.Time) bool {
+	if !IsPodReady(pod) {
```

Comment:
Not in this level. I wanted to keep IsPodAvailable merely as an incremental addition on top of IsPodReady. Callers should decide if they want to filter succeeded/failed/terminating pods, similar to how we use IsPodReady in the replica set and rc controllers.